addon/export.py

```diff
@@ -370,19 +370,21 @@ def export_openscenario(self):
             for obj in bpy.data.collections['OpenDRIVE'].objects:
                 if obj.name.startswith('junction'):
                     if not len(obj['incoming_roads']) == 4:
-                        self.report({'ERROR'}, 'Junction must have 4 connected roads.')
-                        break
+                        self.report({'WARNING'}, 'Junction should have 4 connected roads.')
+                        #break
                     incoming_roads = []
                     angles = []
                     junction_id = obj['id_xodr']
                     # Create junction roads based on incoming road angles (simple 4-way for now)
-                    for idx in range(4):
-                        angles.append(idx * 2 * pi / len(obj['incoming_roads']))
                     # 0 angle road must point in 'right' direction
-                    incoming_roads.append(xodr.get_road_by_id(roads, obj['incoming_roads']['cp_right']))
-                    incoming_roads.append(xodr.get_road_by_id(roads, obj['incoming_roads']['cp_up']))
-                    incoming_roads.append(xodr.get_road_by_id(roads, obj['incoming_roads']['cp_left']))
-                    incoming_roads.append(xodr.get_road_by_id(roads, obj['incoming_roads']['cp_down']))
+                    for idx,cp in zip(range(4),['cp_right','cp_up','cp_left','cp_down']):
+                        try:
+                            inc_road = xodr.get_road_by_id(roads, obj['incoming_roads'][cp])
+                            if(inc_road != None):
+                                incoming_roads.append(inc_road)
+                                angles.append(idx * pi / 2)
+                        except KeyError:
```

Comment:
I am a bit skeptical about this approach of throwing away the exception. Would prefer to catch the error before the exception occurs. (Combined with throwing the warning from above.)